web_monitoring/internetarchive.py

```diff
@@ -421,9 +451,16 @@ def timestamped_uri_to_version(self, dt, uri, *, url,
 
             # If the playback includes a redirect, continue on.
             if res.status_code >= 300 and res.status_code < 400:
+                target = res.headers.get('location')
+                # Wayback sometimes has circular memento redirects ¯\_(ツ)_/¯
+                # TODO: probably need to detect longer loops than just URLs
+                # that redirect to themselves. Need to interrupt every step
+                # along a series of redirects with all this error-handling :(
+                if target == uri:
```

Comment:
Update: since trying to run again with all the latest changes I’ve made here, I’ve already seen several examples where the original URL is a memento, but somewhere along the redirect chain we stop getting playbacks. Here’s one example: 
http://web.archive.org/web/20180316103545id_/http://www.nws.noaa.gov/sp/

So I guess I’d better fix this now :\

Should look into [`Session.resolve_redirects()`](http://docs.python-requests.org/en/master/api/?highlight=redirect#requests.Session.resolve_redirects) for this.

Reply:
There is also [`Response.next()`](http://docs.python-requests.org/en/master/api/?highlight=redirect#requests.Response.next).